pkg/subctl/benchmark/latency.go

```diff
@@ -118,4 +129,30 @@ func runLatencyTest(f *framework.Framework, testParams benchmarkTestParams) {
 
 	framework.By(fmt.Sprintf("Waiting for the client pod %q to exit, returning what client sent", nettestClientPod.Pod.Name))
 	nettestClientPod.AwaitFinishVerbose(Verbose)
+	latencyValues := strings.Split(nettestClientPod.TerminationMessage, "\n")[2]
+	values := strings.Split(latencyValues, ",")
```

Comment:
The table output looks good. The only thing is parsing the output makes it dependent on an expected format which could break if the tool's output was changed. I think we should run `subctl benchmark latency ${KUBECONFIGS_DIR}/kind-config-cluster1 --intra-cluster --verbose` in E2E (after `test_subctl_gather` in _e2e.sh_) to at least catch panics that could occur with incompatible output format. 

Also it still dumps the raw pod output which we don't need now. I think we should change `AwaitFinishVerbose` in shipyard to only print the output if `verbose` is set, ie

```
    if verbose {
        Logf("Pod %q output:\n%s", np.Pod.Name, removeDupDataplaneLines(np.TerminationMessage))
    }
```

Reply:
> A minor nit could be that, since it always only has one row, a vertical table would be more suitable.

Yes. and its simpler to implement. But outputs for all other commands are in this format so implemented this way and won't need any furhter validations for when the tool changes.

> We are making assumptions about the format and content of the output which is fine although it could break if the tool was changed. 

> I think we should change AwaitFinishVerbose in shipyard to only print the output if verbose is set

That would not print any output for `benchmark throughput` if verbose is not passed .

Reply:
@tpantelis How about something like this https://github.com/Jaanki/shipyard/commit/08ce98a919a95e05fef919126fd509b1552f4418 which empties `np.TerminationMessage` if `trim = true` and call `AwaitFinishVerbose` with `trim = true` in benchmark latency. This will just output the table and not raw output

Reply:
The point of no verbosity is to eliminate or minimize output.  I'm not clear on what you mean by "empties np.TerminationMessage" and the purpose of your proposed `trim` flag - it amounts to just printing a blank line if `verbose == false` && `trim == true` .  So it's really not printing anything if `verbose == false` which we can simply accomplish with what I proposed above.

Reply:
Also what about parsing the comma-separated metric names and executing `subctl benchmark` in E2E?

Reply:
`AwaitFinishVerbose` is called by `benchmark throughput` and `benchmark latency`. Removing the else section in 
```
    if verbose {
        Logf("Pod %q output:\n%s", np.Pod.Name, removeDupDataplaneLines(np.TerminationMessage))
    } else {
        fmt.Printf("%s", removeDupDataplaneLines(np.TerminationMessage))
   }
```
 will not print output of `benchmark throughput` (which is in proper format) when `Verbose=false`.

The raw output in `benchmark latency` is coming from
```
else {
        fmt.Printf("%s", removeDupDataplaneLines(np.TerminationMessage))
   }
```
What I am doing with `trim` is skpping this `else` and manually formatting the output by `latencyPrinter.Print(value)` to elimanate raw output and print just the formatted output.

Reply:
> The table output looks good. A minor nit could be that, since it always only has one row, a vertical table would be more suitable.

Like this
```

Minimum Latency Microseconds:	50
Mean Latency Microseconds:	79.09
Maximum Latency Microseconds:	10893
Stddev Latency Microseconds:	88.03
Transaction Rate Tran/s:	12599.844

Minimum Latency Microseconds:	91
Mean Latency Microseconds:	142.28
Maximum Latency Microseconds:	9801
Stddev Latency Microseconds:	124.74
Transaction Rate Tran/s:	7006.188

```

Reply:
Perhaps `benchmark throughput` should pretty print too but that would be another PR.  So we have 3 callers of `AwaitFinishVerbose` and each wants something different:

1. print full verbose output
2. print partial verbose output
3. print no output

Adding yet another param to effect case 3 seems like a hack. We could change `verbose` to a tri-state var, eq `Normal`, `Verbose`, `None` but I think that's overkill.  To me, case 3 should really be `verbose == false` and we should eliminate case 2. So both `latency` and `benchmark` would pass false and `benchmark` would do `fmt.Printf("%s", np.TerminationMessage)`.  We can take that even further and eliminate the `verbose` flag and let the caller decide if and how the output should be printed. I prefer that.

Reply:
`benchmark throughput` is already pretty printed.

```
    if verbose {
        Logf("Pod %q output:\n%s", np.Pod.Name, removeDupDataplaneLines(np.TerminationMessage))
    } else {
        fmt.Printf("%s", removeDupDataplaneLines(np.TerminationMessage))
   }
```

This prints the output from the pod. For others case, it was fine. But for `benchmark latency`, it is not pretty printed. What I am doing is NOT printing the output from the pod directly but formatting it. If `trim=true` and `verbose=false`, do not print the output directly but format it. 
```

This does not change anything for `benchmark throughput` because `trim=false`. But for `benchmark latency`, `trim=true` and hence when `verbose=false`, print just the table and when `verbose=true`, print the raw output + the table

Reply:
The scenario is quite simple:
For `benchmark throughtput`

1. when `verbose=true`, print raw output from
```
if verbose {
        Logf("Pod %q output:\n%s", np.Pod.Name, removeDupDataplaneLines(np.TerminationMessage))
```
2. when verbose=false, print

```
else {
        fmt.Printf("%s", removeDupDataplaneLines(np.TerminationMessage))
   }
```
because the output is already nice and hence `trim=false`

For `benchmark latency`,
1. when `verbose=true`, print raw output from
```
if verbose {
        Logf("Pod %q output:\n%s", np.Pod.Name, removeDupDataplaneLines(np.TerminationMessage))
```
and the table

2. when verbose=false, DO NOT print
```
else {
        fmt.Printf("%s", removeDupDataplaneLines(np.TerminationMessage))
   }
```
because the output is not pretty but instead print the table created in this patch and hence `trim=true`

So `trim` flag basically decided whether to print raw output or formatted output.

Reply:
I understand what you're trying to do. My point is that the `trim` flag as you coded it doesn't make much sense to me.  We don't need to make it more complex by adding another param.  The difference  between `verbose` true and false is that the former also prints the pod name which doesn't seem necessary to me as that's printed elsewhere with `verbose` output. For `benchmark latency`, you're pretty printing it so we don't need it printed twice via `verbose`. `benchmark throughtput` can also print the output as it sees fit. My point is that the `benchmark` commands don't need `AwaitFinishVerbose` to print anything, verbose or not. So we can keep it simple and not muddy `AwaitFinishVerbose` further with a confusing param by doing what I suggested.

Reply:
> For verbose == false, we can simply print the desired output after AwaitFinishVerbose, ie:
> 
 ```
nettestClientPod.AwaitFinishVerbose(Verbose)
 if !Verbose {
     fmt.Print(nettestClientPod.TerminationMessage)
}
```

Shouldn't we check for the pod status before printing?
```
finished := np.Pod.Status.Phase == v1.PodSucceeded || np.Pod.Status.Phase == v1.PodFailed
	if finished {
		np.TerminationCode = np.Pod.Status.ContainerStatuses[0].State.Terminated.ExitCode
                np.TerminationMessage = np.Pod.Status.ContainerStatuses[0].State.Terminated.Message
```

Reply:
yeah `CheckSuccessfulFinish`.